src/DotNetWorker.Grpc/Http/GrpcHttpRequestData.cs

```diff
@@ -20,11 +20,50 @@ internal class GrpcHttpRequestData : HttpRequestData, IAsyncDisposable, IDisposa
         private HttpHeadersCollection? _headers;
         private Stream? _bodyStream;
         private bool _disposed;
+        private Lazy<IReadOnlyCollection<IHttpCookie>> _cookies;
 
         public GrpcHttpRequestData(RpcHttp httpData, FunctionContext functionContext)
             : base(functionContext)
         {
             _httpData = httpData ?? throw new ArgumentNullException(nameof(httpData));
+            _cookies = new Lazy<IReadOnlyCollection<IHttpCookie>>(() =>
+            {
+                if(_headers is null)
+                {
+                    return new List<IHttpCookie>();
+                }
+
+                // this produces either an empty list (no cookie) or a list with a single KeyValuePair
+                // in the format of ("Cookie", "cookie_1=value;cookie_2=value...")
+                var cookieList = _headers.ToLookup(item => item.Key)["Cookie"].ToList();
+
+                if (cookieList.Count > 0)
+                {
+                    // cookieList should have only one KeyValuePair, where the Value is all of the cookies concatenated
+                    // together as a string (ex. "Cookie_1=value;Cookie_2=value;Cookie_3=value")
+                    return cookieList.ConvertAll(new Converter<KeyValuePair<string, IEnumerable<string>>, IReadOnlyCollection<IHttpCookie>>(CookieStringsToHttpCookie)).First();
```

Comment:
Is there a better way to handle the format the cookies come in from the headers? They are all sent in a long string with all the cookies concatenated together.

Reply:
With the proposed change above, you'll end up with the actual string at this point.